mdp/nodes/sfa_nodes.py

```diff
@@ -369,6 +376,170 @@ def get_eta_values(self, t=1):
+                                             rank_deficit_method=rank_deficit_method)
+
+        self.tphase = 0
+
+    def _init_cov(self):
+        # init two covariance matrices
+        # one for the input data
+        self._cov_mtx = VartimeCovarianceMatrix(self.dtype)
+        # one for the derivatives
+        self._dcov_mtx = VartimeCovarianceMatrix(self.dtype)
+
+    def time_derivative(self, x, dt=None):
+        """
+        Compute the linear approximation of the time derivative
+
+        :param x: The time series data.
+        :type x: numpy.ndarray
+
+        :param dt: Sequence of time increments between vectors. Must be
+            of length *x.shape[0]-1* in case of a single training phase.
+            In case the training is done in multiple phases with intended
+            time dependence - meaning the derivative between chunks should
+            be considered, *dt* should be of length *x.shape[0]* starting
+            with the second call. If it continues to be of length
+            *x.shape[0]-1* time dependence is omitted. If *dt* is omitted
+            entirely there will be no time dependence between calls and
+            it will be considered to be one everywhere within a call.
+        :type dt: numpy.ndarray
+
+        :returns: Piecewise linear approximation of the time derivative.
+        :rtype: numpy.ndarray
+        """
+        outlen = x.shape[0] - \
+            1 if self.tphase == 0 or dt is None or dt.shape[0] != x.shape[0] else x.shape[0]
+        out = numx.empty([outlen, x.shape[1]])
+
+        if dt is not None:
+            # Improvements can be made, by interpolating polynomials
+            out[-x.shape[0]+1:, :] = (x[1:, :]-x[:-1, :]) / \
+                dt[-x.shape[0]+1:, None]
+        else:
+            # trivial fallback
+            out[-x.shape[0]+1:, :] = x[1:, :]-x[:-1, :]
+
+        if self.tphase > 0 and dt is not None and dt.shape[0] == x.shape[0]:
+            # check if str before checking, to avoid comparison warnings
+            out[0, :] = numx.divide(x[0, :]-self.xlast, dt[0])
+
+        self.xlast = x[-1, :].copy()
+        self.dtlast = dt[-1].copy() if dt is not None else 1.
+        self.tphase += 1
+
+        return out
+
+    def _train(self, x, dt=None):
+        """
+        Training method.
+
+        :param x: The time series data.
+        :type x: numpy.ndarray
+
+        :param dt: Sequence of time increments between vectors. Must be
+            of length *x.shape[0]-1*. In case the training is done in multiple
+            phases with intended time dependence , *dt* should be of length
+            *x.shape[0]* starting with the second call. If it continues to be
+            of length *x.shape[0]-1* time dependence is omitted. If *dt* is
+            omitted entirely there will be no time dependence between calls and
+            it will be considered to be one everywhere within a call.
```

Comment:
I find the whole explanation hard to understand. Better write it in cases-style:
`len(dt) == x.shape[0]`: `dt[0]` will be used as time increment since previous chunk (ignored on fist call)
`len(dt) == x.shape[0]-1`: No time derivative to previous chunk considered.
`dt is None`: Time increment is assumed as constantly one (like in ordinary SFA).